File: DataAccessLayer/Entities/Photo.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class Photo: BaseEntity
    {
        [Required, StringLength(200)]
        
        public string PhotoPath { get; set; }

        [Required, StringLength(150)]
        public string Title { get; set; }

        [StringLength(300)]
        public string Description { get; set; }

        [Required]
        public DateTime UploadDate { get; set; }

        [Required, DatabaseGenerated(DatabaseGeneratedOption.None)]
        public int UserId { get; set; }

        public double TotalRating { get; set; }

        [AllowNull]
        public User User { get; set; }
        public ICollection<AlbumPhoto> AlbumPhotos { get; set; }
        public ICollection<PhotoRating> PhotoRatings { get; set; }
        public ICollection<PhotoTag> PhotoTags { get; set; }

    }
}

File: UI-Angular/src/app/models/photo.ts
import { raceWith } from "rxjs";

export class Photo{
    id!: number;
    photoPath: string;
    title: string;
    description?: string;
    uploadDate: Date;
    userId: number;
    totalRating: number;
    userName: string;
    photoRatingIds: Array<number>;
    photoIds: Array<number>;
    albumIds: Array<number>;


    constructor(photoPath: string, title: string, description="", uploadDate: Date, userId: number
    , totalRating: number, photoRatingIds: Array<number>, photoIds: Array<number>, albumIds: Array<number>,
    userName: string ){
        this.photoPath = photoPath;
        this.title = title;
        this.description = description;
        this.userId = userId;
        this.uploadDate = uploadDate;
        this.totalRating = totalRating;
        this.photoRatingIds = photoRatingIds;
        this.photoIds = photoIds;
        this.albumIds = albumIds;
        this.userName = userName;
    }
}